federation/pkg/dnsprovider/providers/google/clouddns/rrsets.go

```diff
@@ -43,18 +44,20 @@ func (rrsets ResourceRecordSets) List() ([]dnsprovider.ResourceRecordSet, error)
 }
 
 func (rrsets ResourceRecordSets) Get(name string) (dnsprovider.ResourceRecordSet, error) {
-	var newRrset dnsprovider.ResourceRecordSet
-	rrsetList, err := rrsets.List()
+	response, err := rrsets.impl.List(rrsets.project(), rrsets.zone.impl.Name()).Name(name).Do()
 	if err != nil {
 		return nil, err
 	}
-	for _, rrset := range rrsetList {
-		if rrset.Name() == name {
-			newRrset = rrset
-			break
+	list := make([]dnsprovider.ResourceRecordSet, 0)
+	for _, rrset := range response.Rrsets() {
+		if strings.TrimSuffix(name, ".") == strings.TrimSuffix(rrset.Name(), ".") {
```

Comment:
This trim logic occurs in several different places, and should be factored out into a common utility function, with a comment to explain why it's necessary.